src/interface/src/app/treatments/treatment-map/treatment-map.component.ts

```diff
@@ -200,6 +200,15 @@ export class TreatmentMapComponent {
 
   mapLoaded(event: MapLibreMap) {
     this.mapLibreMap = event;
+    this.listenForZoom();
+  }
+
+  listenForZoom() {
+    this.mapLibreMap.on('zoom', () => {
+      const zoomLevel = this.mapLibreMap.getZoom();
+      // Hide project areas label when we zoom out
+      this.treatmentsState.projectAreaLabelsSize$.next(zoomLevel > 9 ? 14 : 0);
```

Comment:
I have a suggestion here. Instead of doing this on  this.treatmentsState.projectAreaLabelsSize, can you add an observable / behavior subject on mapConfigState that stores the current zoom level?

that way, this component will save the zoom level, and we can decouple the project area label sizes thing.

On map project areas, instead of this:
  ```
  textSize$ = this.treatmentsState.projectAreaLabelsSize$;
  ```
  
you can have something like

```
  textSize$ = this.mapConfigState.zoomLevel$.pipe(map( zoomLevel => zoomLevel > 9 ? 14 : 0)
  ```

Reply:
Updated